### PR TITLE
test: move ln tests to top level bats

### PR DESCRIPTION
**Pull request by sandipndev**
Moves LN tests to top level bats

TODO: Make sure balance_for_check is readded

### CODE REVIEW

File: bats/core/api/ln-receive.bats [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+load "../../helpers/cli.bash"
+load "../../helpers/user.bash"
+load "../../helpers/onchain.bash"
+load "../../helpers/ln.bash"
+load "../../helpers/wallet.bash"
+load "../../helpers/callback.bash"
+load "../../helpers/subscriber.bash"
+
+setup_file() {
+  lnds_init
```
Comment (bodymindarts) [inline bats/core/api/ln-receive.bats:12]: Initialising the lightning setup should be done as part of tilt up IMO

File: bats/core/api/ln-receive.bats [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  create_user 'alice'
+  user_update_username 'alice'
+  fund_user_onchain 'alice' 'btc_wallet'
+  fund_user_onchain 'alice' 'usd_wallet'
+  add_callback 'alice'
+
+  fund_user_lightning 'alice' 'btc_wallet' '500000'
+
+  subscribe_to 'alice' my-updates-sub
+  sleep 3
```
Comment (bodymindarts) [inline bats/core/api/ln-receive.bats:23]: Why sleep 3? Can we have a loop to detect when we're done. Sleep tends to be flaky.

File: bats/core/api/ln-receive.bats [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+  # Check for subscriber event
+  check_for_ln_update "$payment_hash" || exit 1
+}
+
+@test "ln-receive: settles btc-wallet invoices created while trigger down" {
+  token_name="$ALICE_TOKEN_NAME"
+  btc_wallet_name="alice.btc_wallet_id"
+
+  # Stop trigger
+  # stop_trigger
```
Comment (bodymindarts) [inline bats/core/api/ln-receive.bats:256]: Comment?

File: bats/core/api/ln-receive.bats [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+  invoice="$(graphql_output '.data.lnNoAmountInvoiceCreate.invoice')"
+
+  payment_request="$(echo $invoice | jq -r '.paymentRequest')"
+  [[ "${payment_request}" != "null" ]] || exit 1
+  payment_hash="$(echo $invoice | jq -r '.paymentHash')"
+  [[ "${payment_hash}" != "null" ]] || exit 1
+
+  # Start trigger
+  # start_trigger
+  sleep 5
```
Comment (bodymindarts) [inline bats/core/api/ln-receive.bats:274]: sleep?

File: bats/core/api/ln-receive.bats [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+  payment_request="$(echo $invoice | jq -r '.paymentRequest')"
+  [[ "${payment_request}" != "null" ]] || exit 1
+  payment_hash="$(echo $invoice | jq -r '.paymentHash')"
+  [[ "${payment_hash}" != "null" ]] || exit 1
+
+  # Pay invoice
+  lnd_outside_cli payinvoice -f \
+    --pay_req "$payment_request" \
+    --amt "$btc_amount" \
+    &
```
Comment (bodymindarts) [inline bats/core/api/ln-receive.bats:340]: why is this backgrounded?

File: bats/helpers/ln.bash [hunk 6]
```diff
@@ -1,6 +1,120 @@
...
+  [[ "${payment_hash}" != "null" ]]
+
+  lnd_outside_cli payinvoice -f \
+    --pay_req "$payment_request" \
+    --amt "$amount"
+
+  retry 15 1 check_for_ln_initiated_settled "$token_name" "$payment_hash"
+}
+
+lnds_init() {
```
Comment (bodymindarts) [inline bats/helpers/ln.bash:31]: This should probably be a setup step in the tilt file.

Comment (vindard) [reply bats/helpers/ln.bash:31]: Added the file here #3718

File: bats/core/api/ln-receive.bats (hunk 1 above)
Comment (vindard) [reply bats/core/api/ln-receive.bats:12]: #3718

**Pull request closed without merging**